File: typescript/examples/_shared/items.ts
import type { ItemOut, ItemsApi } from "cs2cap-sdk";

export async function resolveCatalogItems(
  itemsApi: ItemsApi,
  options: {
    query?: string;
    itemType?: string;
    limit?: number;
  } = {},
): Promise<ItemOut[]> {
  const response = await itemsApi.listItemsV1ItemsGet({
    q: options.query,
    itemType: options.itemType,
    limit: options.limit ?? 10,
  });
  return [...response.items];
}

export function requireItemId(item: ItemOut): number {
  if (item.itemId == null) {
    throw new Error(`Item ${JSON.stringify(item.marketHashName)} did not include an itemId.`);
  }
  return item.itemId;
}

export function collectItemIds(items: Iterable<ItemOut>): number[] {
  return Array.from(items, requireItemId);
}

File: python/examples/_shared/items.py
from __future__ import annotations

from collections.abc import Iterable

import cs2cap


def resolve_catalog_items(
    items_api: cs2cap.ItemsApi,
    *,
    query: str | None = None,
    item_type: str | None = None,
    limit: int = 10,
) -> list[cs2cap.ItemOut]:
    response = items_api.list_items(
        q=query,
        item_type=item_type,
        limit=limit,
    )
    return list(response.items)


def require_item_id(item: cs2cap.ItemOut) -> int:
    if item.item_id is None:
        raise ValueError(f"Item {item.market_hash_name!r} did not include an item_id.")
    return item.item_id


def collect_item_ids(items: Iterable[cs2cap.ItemOut]) -> list[int]:
    return [require_item_id(item) for item in items]
